src/app/campaign/flight/flight-form-control-container.component.ts

```diff
@@ -128,42 +119,8 @@ export class FlightFormControlContainerComponent implements OnInit, OnDestroy {
     }
   }
 
-  get zonesControls(): FormArray {
-    return this.flightForm.get('zones') as FormArray;
-  }
-
-  createZoneControl() {
-    return this.fb.group({ id: ['', Validators.required], url: ['', validateMp3] });
-  }
-
-  onAddZone() {
-    // create new form group/zone control with initial empty values
-    const newZone = this.createZoneControl();
```

Comment:
This was causing two update cycles. One with the empty zone, and second with the control reset to the first option. This is one of the things that led me down the unidirectional data flow way of thinking. Maybe the intention was not to cause an update cycle at all by manipulating the form?